webapp/plugins/index.js

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+// EXPERIMENTAL - SUBJECT TO CHANGE
+
+import store from 'stores/redux_store.jsx';
+
+global.window.plugins = {};
+global.window.plugins.components = {};
+
+export function registerComponents(components) {
+    global.window.plugins.components = {...components, ...global.window.plugins.components};
+}
+
+export function initializePlugins() {
+    const pluginJson = global.window.mm_config.Plugins || '[]';
+    const pluginManifests = JSON.parse(pluginJson);
+
+    pluginManifests.forEach((m) => {
+        // Fetch the plugin's bundled js
+        const xhrObj = new XMLHttpRequest();
+        xhrObj.open('GET', m.bundle_path, false);
```

Comment:
I don't think we want to support getting the bundle from anywhere. Only from the MM server. Is this temporary?